spec/features/create_preassembly_image_spec.rb

```diff
@@ -129,8 +131,43 @@
     yaml = YAML.load_file(download)
     expect(yaml[:status]).to eq 'success'
 
-    visit "#{Settings.argo_url}/view/druid:#{yaml[:pid]}"
-
-    reload_page_until_timeout!(text: "v#{version + 1} Accessioned", with_reindex: true)
+    prefixed_druid = "druid:#{yaml[:pid]}"
+    latest_version = version + 1
+
+    visit "#{Settings.argo_url}/view/#{prefixed_druid}"
+    reload_page_until_timeout!(text: "v#{latest_version} Accessioned", with_reindex: true)
+
+    # The below confirms that preservation replication is working: we only replicate a
+    # Moab version once it's been written successfully to on prem storage roots, and
+    # we only log an event to dor-services-app after a version has successfully replicated
+    # to a cloud endpoint.  So, confirming that both versions of our test object have
+    # replication events logged for all three cloud endpoints is a good basic test of the
+    # entire preservation flow.
+
+    # hacky, give pres cat replication pipeline a chance to finish
+    sleep 30
+
+    druid_tree_str = DruidTools::Druid.new(prefixed_druid).tree.join('/')
+
+    latest_s3_key = "#{druid_tree_str}.v000#{latest_version}.zip"
+    reload_page_until_timeout!(text: latest_s3_key, with_events_expanded: true)
+
+    # the event log should eventually contain an event for replication of each version that
+    # this test created to every endpoint we archive to
+    poll_for_matching_events!(prefixed_druid) do |events|
```

Comment:
I wonder if it makes sense for the helper method to be closer to "poll_for_event_matching(druid, version, something)".   Or not.  Just a thought, certainly not a blocker.

Reply:
i somewhat prefer it as is, because it allows the polling logic to be re-used easily, while keeping the logic about what events are relevant in the specific test that's invoking the polling.  though i can also see the appeal of hiding away the logic for which events match.